src/main/io/serial.c

```diff
@@ -574,39 +574,29 @@ void closeSerialPort(serialPort_t *serialPort)
 
 void serialInit(bool softserialEnabled, serialPortIdentifier_e serialPortToDisable)
 {
-#if !defined(USE_SOFTSERIAL)
-    UNUSED(softserialEnabled);
-#endif
-
     memset(&serialPortUsageList, 0, sizeof(serialPortUsageList));
 
     for (int index = 0; index < SERIAL_PORT_COUNT; index++) {
         serialPortUsageList[index].identifier = serialPortIdentifiers[index];
 
-        if (serialPortToDisable != SERIAL_PORT_NONE
-            && serialPortUsageList[index].identifier == serialPortToDisable) {
-                serialPortUsageList[index].identifier = SERIAL_PORT_NONE;
-                continue;  // this index is deleted
+        if (serialPortToDisable != SERIAL_PORT_NONE && serialPortUsageList[index].identifier == serialPortToDisable) {
+            // this index is deleted
```

Comment:
(old comment was meant for `continue` only, ale it was mostly leftover from before refactoring.
```suggestion
            // disable port
```